source/common/network/address_impl.h

```diff
@@ -252,7 +261,8 @@ class Ipv6Instance : public InstanceBase {
    * upon error.
    */
   Ipv6Instance(absl::Status& error, const sockaddr_in6& address, bool v6only = true,
-               const SocketInterface* sock_interface = nullptr);
+               const SocketInterface* sock_interface = nullptr,
+               absl::optional<std::string> network_namespace = absl::nullopt);
```

Comment:
I think you to modify friendly_name or addresstostring - there are several hidden places in Envoy where addresses are mapped by their string name, so we need to have unique representations.